### PR TITLE
Bug 1797894: Add MachineNetworks to proxy.status.noPoxy

### PR DESCRIPTION
**Pull request by juanluisvaladas**
.

### CODE REVIEW

**Comment by juanluisvaladas** [comment]
I added  github.com/openshift/installer as a dependency pointing to the master branch.
Should be safe because we're only using the ipnet.IPNet struct.

All the code changes are in pkg/util/proxyconfig/no_proxy.go
The rest are changes to Gopgk.* and adding github.com/openshift/installer as a dependency.

File: Gopkg.toml [hunk 1]
```diff
@@ -60,6 +60,10 @@ required = [
   branch = "master"
   name = "github.com/openshift/api"
 
+[[constraint]]
+  branch = "master"
+  name = "github.com/openshift/installer"
```
Comment (danwinship) [inline Gopkg.toml]: we should not be vendoring installer

File: pkg/util/proxyconfig/no_proxy.go [hunk 2]
```diff
@@ -21,14 +22,20 @@ import (
 // are returned.
 func MergeUserSystemNoProxy(proxy *configv1.Proxy, infra *configv1.Infrastructure, network *configv1.Network, cluster *corev1.ConfigMap) (string, error) {
 	// TODO: This will be flexible when master machine management is more dynamic.
+	type machineNetworkEntry struct {
+		// CIDR is the IP block address pool for machines within the cluster.
+		CIDR ipnet.IPNet `json:"cidr"`
```
Comment (danwinship) [inline pkg/util/proxyconfig/no_proxy.go]: Just make this a `string` and parse it yourself

**Comment by juanluisvaladas** [comment]
/hold
haven't actually tested this yet

**Comment by juanluisvaladas** [comment]
I'm doing the cherry pick before this is merged because this is blocking some people. Hopefully I won't need more changes so this will save some time.
/cherry-pick release-4.4
/cherry-pick release-4.3

**Comment by danwinship** [comment]
> I'm doing the cherry pick before this is merged...

> @juanluisvaladas: once the present PR merges, I will cherry-pick it on top of release-4.4

DENIED!

**Pull request merged**